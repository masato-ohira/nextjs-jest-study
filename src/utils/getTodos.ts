import axios from 'axios'

type TodoType = {
  userId: number
  id: number
  title: string
  completed: boolean
}

export const getTodos = async (): Promise<TodoType[]> => {
  const { data } = await axios.get('/api/todos')
  return data
}
