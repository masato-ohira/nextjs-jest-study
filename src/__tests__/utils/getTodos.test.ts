import { getTodos } from '@/utils/getTodos'
import { map } from 'lodash-es'

test('getTodosの型は常に同じである', async () => {
  const todos = await getTodos()
  const todoKeys = map(todos[0], (val, key) => key)
  console.log({ todoKeys })

  expect(todoKeys).toMatchSnapshot()
})
