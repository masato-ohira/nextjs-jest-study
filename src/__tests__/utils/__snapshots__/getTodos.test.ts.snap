// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTodosの型は常に同じである 1`] = `
[
  "userId",
  "id",
  "title",
  "completed",
]
`;
